ie_serving/server/predict_utils.py

```diff
@@ -117,21 +117,21 @@ def prepare_output_as_list(inference_output, model_available_outputs):
 
 
 '''
-The function is not used.
-Probably preparing the output would be faster,
-but you need a change of grpc clients.
+The function prepare_output_with_tf implements faster serialization mechanizm
+It returns slightly different format of output but it can be processed
+by make_ndarray on the client side the same way like with
+prepare_output_as_list
+'''
+
 
 def prepare_output_with_tf(inference_output, model_available_outputs):
     response = predict_pb2.PredictResponse()
-
     for output in model_available_outputs:
+        model_output = model_available_outputs[output]
         response.outputs[output].CopyFrom(
-            tf_contrib_util.make_tensor_proto(inference_output[output],
-                                              shape=inference_output[output].
-                                              shape,
-                                              dtype=dtypes.as_dtype(
-                                                  inference_output
-                                                  [output].dtype).
-                                              as_datatype_enum))
+            tf_contrib_util.make_tensor_proto(
+                inference_output[model_output],
+                shape=inference_output[model_output].shape,
+                dtype=dtypes.as_dtype(inference_output[model_output].dtype).

```

Comment:
shape and dtype will be used from `inference_output[model_output]` by default, so `tf_contrib_util.make_tensor_proto(inference_output[model_output])` must be enough